### PR TITLE
Fixed a broken test

### PR DESCRIPTION
**Pull request by tmatsuo**

### CODE REVIEW

**Comment by tmatsuo** [comment]
Also running the tests on our jenkins

**Comment by tmatsuo** [comment]
`appengine/flexible/metadata/test/DeployTest.php`, it must have been broken for a while. I think it happened with refactoring. The fix in `datastore/tutorial/test/FunctionsTest.php` is for less flakiness.

**Comment by tmatsuo** [comment]
The test is still failing with:

```
1) LocalTest::testHome
Google\Cloud\Exception\ServiceException: {
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "forbidden",
    "message": "Caller does not have storage.objects.get access to object cloud-samples-tests-php/hello_flex.txt."
   }
  ],
  "code": 403,
  "message": "Caller does not have storage.objects.get access to object cloud-samples-tests-php/hello_flex.txt."
 }
}


/workspace/php-docs-samples/appengine/flexible/storage/vendor/google/cloud/src/RequestWrapper.php:219
/workspace/php-docs-samples/appengine/flexible/storage/vendor/google/cloud/src/RequestWrapper.php:136
/workspace/php-docs-samples/appengine/flexible/storage/vendor/google/cloud/src/RestTrait.php:80
/workspace/php-docs-samples/appengine/flexible/storage/vendor/google/cloud/src/Storage/Connection/Rest.php:173
/workspace/php-docs-samples/appengine/flexible/storage/vendor/google/cloud/src/Storage/StorageObject.php:135
/workspace/php-docs-samples/appengine/flexible/storage/app.php:35
/workspace/php-docs-samples/appengine/flexible/storage/vendor/symfony/http-kernel/HttpKernel.php:144
/workspace/php-docs-samples/appengine/flexible/storage/vendor/symfony/http-kernel/HttpKernel.php:64
/workspace/php-docs-samples/appengine/flexible/storage/vendor/silex/silex/src/Silex/Application.php:586
/workspace/php-docs-samples/appengine/flexible/storage/vendor/symfony/http-kernel/Client.php:79
/workspace/php-docs-samples/appengine/flexible/storage/vendor/symfony/browser-kit/Client.php:315
/workspace/php-docs-samples/appengine/flexible/storage/test/LocalTest.php:56
```

**Comment by tmatsuo** [comment]
It fails when the file in question was uploaded by another service account, because the service account used in the test can not read that file.

I have just updated our Travis config to use the same service account as the one used on our jenkins.

**Comment by tmatsuo** [comment]
Still failing:

```
There was 1 error:

1) Google\Cloud\Samples\Kms\KeyCommandTest::testListCryptoKeys
Google_Service_Exception: {
  "error": {
    "code": 400,
    "message": "Resource name 'projects/cloud-samples-tests-php/locations/global/keyRings/' does not match pattern 'projects/([^/]+)/locations/([a-zA-Z0-9_-]{1,63})/keyRings/([a-zA-Z0-9_-]{1,63})'.",
    "errors": [
      {
        "message": "Resource name 'projects/cloud-samples-tests-php/locations/global/keyRings/' does not match pattern 'projects/([^/]+)/locations/([a-zA-Z0-9_-]{1,63})/keyRings/([a-zA-Z0-9_-]{1,63})'.",
        "domain": "global",
        "reason": "badRequest"
      }
    ],
    "status": "INVALID_ARGUMENT"
  }
}


/workspace/php-docs-samples/kms/api/vendor/google/apiclient/src/Google/Http/REST.php:118
/workspace/php-docs-samples/kms/api/vendor/google/apiclient/src/Google/Http/REST.php:94
/workspace/php-docs-samples/kms/api/vendor/google/apiclient/src/Google/Task/Runner.php:181
/workspace/php-docs-samples/kms/api/vendor/google/apiclient/src/Google/Http/REST.php:58
/workspace/php-docs-samples/kms/api/vendor/google/apiclient/src/Google/Client.php:789
/workspace/php-docs-samples/kms/api/vendor/google/apiclient/src/Google/Service/Resource.php:232
/workspace/php-docs-samples/kms/api/vendor/google/apiclient-services/src/Google/Service/CloudKMS/Resource/ProjectsLocationsKeyRingsCryptoKeys.php:130
/workspace/php-docs-samples/kms/api/src/functions.php:397
/workspace/php-docs-samples/kms/api/src/KeyCommand.php:106
/workspace/php-docs-samples/kms/api/vendor/symfony/console/Command/Command.php:262
/workspace/php-docs-samples/kms/api/vendor/symfony/console/Tester/CommandTester.php:86
/workspace/php-docs-samples/kms/api/test/KeyCommandTest.php:51
```

@bshaffer Can you look at it and add a commit to this branch?

**Comment by tmatsuo** [comment]
It looks like there are no KMS related envvar on our jenkins. I change the test to skip when the envvar is not present.

**Comment by tmatsuo** [comment]
Travis seems busy. I kicked off our jenkins build.

**Comment by tmatsuo** [comment]
The tests are still failing. Note: there must be one deployment is leaking in one of our e2e tests, and there were lots of versions deployed. I'm cleaning them up, and run the tests again.

File: kms/api/test/KeyCommandTest.php [hunk 1]
```diff
@@ -32,7 +32,8 @@ public function setUp()
         if (!$projectId = getenv('GOOGLE_PROJECT_ID')) {
             return $this->markTestSkipped('Set the GOOGLE_PROJECT_ID environment variable');
         }
-
+        if (!$ring = getenv('GOOGLE_KMS_KEYRING')) {
+            return $this->markTestSkipped('Set the GOOGLE_KMS_KEYRING environment variable');
         $this->projectId = $projectId;
         $this->ring = getenv('GOOGLE_KMS_KEYRING');
```
Comment (bshaffer) [inline kms/api/test/KeyCommandTest.php]: Should be `$this->ring = $ring`

**Comment by tmatsuo** [comment]
Getting better, but it's still not green.

```
running phpunit in kms/api
+ phpunit
PHPUnit 4.8.24 by Sebastian Bergmann and contributors.
Warning:	The Xdebug extension is not loaded
		No code coverage will be generated.

EEESESESES..........ES

Time: 3.77 seconds, Memory: 17.75Mb

There were 7 errors:

1) Google\Cloud\Samples\Kms\EncryptionCommandTest::testEncrypt
Google_Service_Exception: {
  "error": {
    "code": 403,
    "message": "The caller does not have permission",
    "errors": [
      {
        "message": "The caller does not have permission",
        "domain": "global",
        "reason": "forbidden"
      }
    ],
    "status": "PERMISSION_DENIED"
  }
}
```

@bshaffer If you can take over this CL, that would be great since you're the most knowledgeable on those tests.

**Pull request merged**